src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -386,15 +390,15 @@ ClassFileStream* ClassPathImageEntry::open_stream(Thread* current, const char* n
 //
 ClassFileStream* ClassPathImageEntry::open_stream_for_loader(Thread* current, const char* name, ClassLoaderData* loader_data) {
   jlong size;
-  JImageLocationRef location = (*JImageFindResource)(_jimage, "", get_jimage_version_string(), name, &size);
+  JImageLocationRef location = (*JImageFindResource)(jimage(), "", get_jimage_version_string(), name, &size);
```

Comment:
There's a subtle change:

* before: `_jimage` could never be NULL, since it's a private member of this `ClassPathImageEntry`
* after: `jimage()` could theoretically be NULL, if `ClassPathImageEntry::close_jimage()` is called, which happens only when:
  * `~ClassPathImageEntry()` is called. But this will never happen because we never delete `ClassPathImageEntry`;
  * `ClassLoader::close_jrt_image()` is called. This happens only on Linux inside `os::abort()`

When we call `jimage()` and expect it to be non-null, maybe we should use a new function `jimage_non_null()` like this:

```
JImageFile* ClassPathImageEntry::jimage_non_null() {
  assert(ClassLoader::has_jrt_entry(), "must be");
  assert(jimage() != NULL, "should have been opened by ClassLoader::lookup_vm_options "
                            "and remained throughout normal JVM lifetime");
  return jimage();
}
```

and we should change `~ClassPathImageEntry()` to assert that it's never called?